vowpalwabbit/reductions/cb/cb_adf.cc

```diff
@@ -274,42 +307,12 @@ void cb_adf::learn_MTR(multi_learner& base, VW::multi_ex& examples)
   }
 }
 
-// Validates a multiline example collection as a valid sequence for action dependent features format.
-VW::example* test_adf_sequence(const VW::multi_ex& ec_seq)
-{
-  if (ec_seq.empty()) THROW("cb_adf: At least one action must be provided for an example to be valid.");
-
-  uint32_t count = 0;
-  VW::example* ret = nullptr;
-  for (auto* ec : ec_seq)
-  {
-    // Check if there is more than one cost for this example.
-    if (ec->l.cb.costs.size() > 1)
-    {
-      auto message = fmt::format(
-          "cb_adf: badly formatted example, only one cost can be known but found {}. Example number={}, tag={}",
-          ec->l.cb.costs.size(), ec->example_counter, VW::string_view{ec->tag.data(), ec->tag.size()});
-      THROW(message);
-    }
-
-    // Check whether the cost was initialized to a value.
-    if (ec->l.cb.costs.size() == 1 && ec->l.cb.costs[0].cost != FLT_MAX)
-    {
-      ret = ec;
-      count += 1;
-      if (count > 1) THROW("cb_adf: badly formatted example, only one line can have a cost");
-    }
-  }
-
-  return ret;
-}
-
 void cb_adf::learn(multi_learner& base, VW::multi_ex& ec_seq)
 {
-  if (test_adf_sequence(ec_seq) != nullptr)
+  if (CB_ADF::test_adf_sequence(ec_seq) != nullptr)
```

Comment:
not related to this PR but it is unfortunate that we have `CB_ADF` and `cb_adf` namespaces

Reply:
It looks like `cb_adf` is just the class name